packages/wallets/bravewallet/src/adapter.ts

```diff
@@ -0,0 +1,268 @@
+                            resolve();
+                        };
+
+                        wallet._handleDisconnect = (...args: unknown[]) => {
+                            wallet.off('connect', connect);
+                            reject(new WalletWindowClosedError());
+                            return handleDisconnect.apply(wallet, args);
+                        };
+
+                        wallet.on('connect', connect);
+
+                        wallet.connect().catch((reason: any) => {
+                            wallet.off('connect', connect);
+                            reject(reason);
+                        });
+                    });
+                } catch (error: any) {
+                    if (error instanceof WalletError) throw error;
+                    throw new WalletConnectionError(error?.message, error);
+                } finally {
+                    wallet._handleDisconnect = handleDisconnect;
+                }
+            }
+
+            if (!wallet.publicKey) throw new WalletAccountError();
+
+            let publicKey: PublicKey;
+            try {
+                publicKey = new PublicKey(wallet.publicKey.toBytes());
+            } catch (error: any) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+
+            wallet.on('disconnect', this._disconnected);
+
+            this._wallet = wallet;
+            this._publicKey = publicKey;
+
+            this.emit('connect', publicKey);
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        } finally {
+            this._connecting = false;
+        }
+    }
+
+    async disconnect(): Promise<void> {
+        const wallet = this._wallet;
+        if (wallet) {
+            wallet.off('disconnect', this._disconnected);
+
+            this._wallet = null;
+            this._publicKey = null;
+
+            try {
+                await wallet.disconnect();
+            } catch (error: any) {
+                this.emit('error', new WalletDisconnectionError(error?.message, error));
+            }
+        }
+
+        this.emit('disconnect');
+    }
+
+    async sendTransaction(
+        transaction: Transaction,
+        connection: Connection,
+        options?: SendTransactionOptions
+    ): Promise<TransactionSignature> {
+        try {
+            const wallet = this._wallet;
+            // Brave doesn't handle partial signers, so if they are provided, don't use `signAndSendTransaction`
+            if (wallet && 'signAndSendTransaction' in wallet && !options?.signers) {
+                // HACK: Brave's `signAndSendTransaction` should always set these, but doesn't yet
+                transaction.feePayer = transaction.feePayer || this.publicKey || undefined;
+                transaction.recentBlockhash =
+                    transaction.recentBlockhash || (await connection.getRecentBlockhash('finalized')).blockhash;
```

Comment:
Same here, please don't copy the Phantom adapter's workarounds. Partially signed transactions should also be supported.

Reply:
fixed and squashed.

Reply:
@jordansexton Is there a documentation we could reference for how partially signed transactions should be supported? What's the expect behavior for the wallet, like what should we do with options.signers? As I understand for cases like signTransaction and signAllTransactions, some signatures can be provided via transaction object's signatures property, I assume it would be the same when using signAndSendTransaction, and I'm not sure what should we do with options.signers.

Reply:
Hmm, specific documentation might be hard to come by, but here's how we implement it in the base class:

https://github.com/solana-labs/wallet-adapter/blob/244af02cb3dcf9334abf61cb9a11e497bde86d57/packages/core/base/src/signer.ts#L25-L33

The reason we have this is to support program / multisig wallets that need to accept meta-transactions, create a new transaction, then sign it with the partial signers provided.

I think in Brave's case this doesn't matter, so you can just accept the array of optional signers, and if provided, sign with them along with the user.